src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -409,10 +410,27 @@ private function describeValue($value, bool $omitTags, bool $showArguments, Cont
         }
 
         if ($value instanceof Reference) {
-            return [
-                'type' => 'service',
-                'id' => (string) $value,
-            ];
+            try {
+                $node = $container?->getCompiler()?->getServiceReferenceGraph()?->getNode((string) $value);
+
+                if ($withAliases && $node?->getValue()?->getClass()) {
+                    return [
+                        'alias' => $node->getValue()->getClass(),
+                        'type' => 'service',
+                        'id' => (string) $value,
+                    ];
+                } else {
+                    return [
+                        'type' => 'service',
+                        'id' => (string) $value,
+                    ];
+                }
+            } catch (\Throwable $throwable) {
```

Comment:
can this be more specific? what kind of exceptions can we expect here?

Reply:
@jlslew Any feedback?